src/inferencer/constraintStore.ts

```diff
@@ -75,9 +88,28 @@ function solveConstraint(constraintLhs: Type, constraintRhs: Type): any | undefi
     constraintStore.get(constraintLhs) === undefined &&
     constraintStore.get(constraintRhs) !== undefined
   ) {
+    // Rule 7B
+    if (
+      constraintStore.get(constraintLhs) !== undefined &&
+      (constraintStore.get(constraintLhs) as Variable).isAddable &&
+      constraintStore.get(constraintRhs) !== undefined &&
+      isTypeVariable(constraintStore.get(constraintRhs))
```

Comment:
For this, we may need to add one more part:
`!constraintStore.get(constraintRhs).isAddable` 
to check that it is a regular type variable